corruption_t5.py

```diff
@@ -500,6 +529,7 @@ def corrupt_for_vanilla_t5(
         pmi: Whether to use PMI-based corruption.
         ngram_vocab_set: The set of ngrams to use for PMI-based corruption.
         tokenizer: The tokenizer to use for PMI-based corruption.
+        seed: The seed to use for random number generation.
```

Comment:
No need for the seed in this part of the code. Within the test, set the seed during 'setup()', or at the beginning of the unit test. You can even pass in a seed as part of a parameterized unit tests. You should not set the seed or deal with logic surrounding it within the function itself.

Reply:
I feel like this should depend on the way we decide to seed.
https://stackoverflow.com/questions/65685060/unique-seed-acrossing-multiple-imported-files-with-random-module-python
According to this answer, in the native `random` module, for random calls to be deterministic across different files, you need to pass the random seed. To avoid that, you can just set it at each file.

To avoid it all together, the `random` module should not be used and should be switched for something else

Reply:
See CorruptionTest in corrruption_t5_test.py. You'll see there that if you set the seed at the start of the test, you will get deterministic behavior. In those unit tests, we compare corruption functions on specific tensor inputs, and expect to get back specific tensor outputs.

https://github.com/zbambergerNLP/ml_training/blob/master/corruption_t5_test.py#L39-L91